src/app/components/RoundedBalance/__tests__/index.test.tsx

```diff
@@ -4,45 +4,69 @@ import { RoundedBalance } from './../'
 describe('RoundedBalance', () => {
   it('should keep original value', () => {
     render(<RoundedBalance value="0.00075" ticker="USDC" />)
-
     expect(screen.getByText('0.00075 USDC')).toBeInTheDocument()
   })
 
-  it('should render value', () => {
-    render(<RoundedBalance value="0.002231" ticker="USDC" />)
-
+  it('should render rounded value', () => {
+    render(<RoundedBalance value="0.00223100001" ticker="USDC" />)
     expect(screen.getByText('0.00223… USDC')).toBeInTheDocument()
   })
 
-  it('should render rounded value', () => {
-    render(<RoundedBalance value="0.00223100001" ticker="USDC" />)
+  it('should render sparse decimals', () => {
+    render(<RoundedBalance value="0.10000000001" ticker="USDC" />)
+    expect(screen.getByText('0.10000… USDC')).toBeInTheDocument()
+  })
 
-    expect(screen.getByText('0.00223… USDC')).toBeInTheDocument()
+  it('should render sparse whole number', () => {
+    render(<RoundedBalance value="100.00000000001" ticker="USDC" />)
+    expect(screen.getByText('100.00000… USDC')).toBeInTheDocument()
   })
 
   it('should render variant when rounded value is equal zero', () => {
     render(<RoundedBalance value="0.00000000000002231" ticker="USDC" />)
-
     expect(screen.getByText('< 0.00000 USDC')).toBeInTheDocument()
   })
 
   it('should not round value up', () => {
     render(<RoundedBalance value="0.004795600000000000" ticker="USDC" />)
-
     expect(screen.getByText('0.00479… USDC')).toBeInTheDocument()
   })
 
   it('should not round negative value up', () => {
     render(<RoundedBalance value="-0.004795600000000000" ticker="USDC" />)
-
     expect(screen.getByText('-0.00479… USDC')).toBeInTheDocument()
   })
-})
 
-describe('RoundedRoseBalance', () => {
   it('should render value with ROSE ticker symbol', () => {
     render(<RoundedBalance value="0.002231" ticker="ROSE" />)
-
     expect(screen.getByText('0.00223… ROSE')).toBeInTheDocument()
   })
+
+  describe('should render large values', () => {
+    it('whole number', () => {
+      render(<RoundedBalance value="1111111111111111111111111111111111111111111111111111" ticker="ROSE" />)
+      expect(
+        screen.getByText('1,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111 ROSE'),
+      ).toBeInTheDocument()
+    })
+
+    it('with decimals', () => {
+      render(
+        <RoundedBalance
+          value="1111111111111111111111111111111111111111111111111111.00000000000002231"
+          ticker="ROSE"
+        />,
+      )
+      expect(
+        screen.getByText('1,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111,111.00000… ROSE'),
```

Comment:
This is improvement, but (using more readable numbers) shouldn't we have something like: 
```
    it('whole number', () => {
      render(<RoundedBalance value="15000000.00000000000002231" ticker="ROSE" />)
      expect(screen.getByText('15M ROSE')).toBeInTheDocument()
    })
```

Reply:
yep; but I don't think we have designs for that anywhere yet. Not sure which components we would want compact notation and which wouldn't

Reply:
if it's not scoped for launch then this looks good